UN-OCHA/hid_api
# HID-2118: Add org/env to OAuth Client config

# HID-2118

We currently have a naming convention on most of our properties that follows: `org - Site name - env` — for a couple reasons we are changing the org/env to be separate fields:

- The system was mostly working but it's getting a bit confusing
- Certain end users were curious what "prod" means and we felt it best to start hiding that info from folks as it sounds strange when you aren't dealing with dev/test/prod all day.

The new fields are available in this PR to be used on an optional basis, allowing a gradual transition as we do data entry. 

## Testing

- Log in as an HID Admin and see if the new fields work as expected on the OAuth Client editing form. You will go to "Admin" in the top dropdown and edit any client, but keep in mind you have to test the Authorize form so try at least one that you have configured locally, e.g. GHO Local.
- Log in as an HID user and use the CD Header dropdown to visit **Settings** where you will see a list of clients. The "x" button now says "Revoke" and it should work like normal. Additionally, after you've edited a few Clients as Admin, those same clients should list any combination of metadata that you entered with one exception: environments of "Production" or "Prod" are not shown to the user.
- Log in via a Drupal site and see if the form mentions that the environment is for development/testing. We do a simple string check and look for "Prod" and failing to find it means we consider it for development.

Some screenshots that hopefully make it clearer:

<img width="645" alt="HID-2118-authorize" src="https://user-images.githubusercontent.com/254753/110655416-53d2e980-81bf-11eb-9211-c28a0fb48bf1.png">
<img width="1121" alt="HID-2118-user-settings" src="https://user-images.githubusercontent.com/254753/110655427-56cdda00-81bf-11eb-9e10-323c15160867.png">
